core/core-impl/src/main/java/io/fabric8/launcher/core/impl/filters/SecuredFilter.java

```diff
@@ -42,29 +56,40 @@ public void filter(ContainerRequestContext requestContext) {
         String token = Authorizations.removeBearerPrefix(authorizationHeader);
 
         try {
-
-            // Validate the token
-            DecodedJWT jwt = validateToken(token);
-            JWTSecurityContext securityContext = new JWTSecurityContext(jwt);
-            // Set the user name as a request property
-            requestContext.setProperty("USER_NAME", securityContext.getUserPrincipal().getName());
-            requestContext.setSecurityContext(securityContext);
+            final DecodedJWT jwt = JWT.decode(token);
+            if (shouldValidate(request)) {
+                validateToken(jwt);
+            }
+            propagateSecurityContext(requestContext, jwt);
 
         } catch (Exception e) {
             abortWithUnauthorized(requestContext);
         }
     }
 
-    private void abortWithUnauthorized(ContainerRequestContext requestContext) {
+    private void propagateSecurityContext(ContainerRequestContext requestContext, DecodedJWT jwt) {
+        final JWTSecurityContext securityContext = new JWTSecurityContext(jwt);
+        requestContext.setProperty("USER_NAME", securityContext.getUserPrincipal().getName());
+        requestContext.setSecurityContext(securityContext);
+    }
 
-        // Abort the filter chain with a 401 status code response
-        requestContext.abortWith(
-                Response.status(Response.Status.UNAUTHORIZED)
-                        .build());
+    // We do not validate tokens in case no keycloak linked for standalone launcher
+    private boolean shouldValidate(HttpServletRequest request) {
```

Comment:
@gastaldi here's how I decide if we should validate or not

Reply:
LGTM